src/api/callback.cc

```diff
@@ -161,11 +161,12 @@ MaybeLocal<Value> InternalMakeCallback(Environment* env,
   Local<Function> hook_cb = env->async_hooks_callback_trampoline();
   int flags = InternalCallbackScope::kNoFlags;
   int hook_count = 0;
+  AsyncHooks* async_hooks = env->async_hooks();
   if (!hook_cb.IsEmpty()) {
     flags = InternalCallbackScope::kSkipAsyncHooks;
-    AsyncHooks* async_hooks = env->async_hooks();
     hook_count = async_hooks->fields()[AsyncHooks::kBefore] +
-                 async_hooks->fields()[AsyncHooks::kAfter];
+                 async_hooks->fields()[AsyncHooks::kAfter] +
+                 async_hooks->fields()[AsyncHooks::kUsesExecutionAsyncResource];
```

Comment:
nit: by adding `kUsesExecutionAsyncResource` the name `hook_name` is misleading.
What about using a bool `hooks_used = (async_hooks->fields()[AsyncHooks::kBefore] + ...) > 0;`?

Reply:
Seconded. Mixing this into `hook_count` is a bit misleading. The _purpose_ of hook_count is just to check if there is any hooks attached though so, as @Flarna said, could change to a bool.